src/target/riscv32.c

```diff
@@ -141,6 +150,45 @@ static void riscv32_regs_write(target_s *const target, const void *const data)
 	riscv_csr_write(hart, RV_DPC, &regs->pc);
 }
 
+static ssize_t riscv32_bool_to_4(bool ret)
+{
+	if (ret)
```

Comment:
Stylistically this is fine, but a suggestion as this can only ever return two values: Use a ternary as in `return ret ? 4 : -1;`. It does make the body a one-line body but there's just no point here to making it an expandable if form piece of conditional logic as `bool` only ever has two states.

Also, please mark the parameter `const`. Same with the other functions introduced - please mark their read-only parameters const and pointers (but not the things those pointers point to if written) const.